app/controllers/agent/AgentHubController.scala

```diff
@@ -16,39 +16,70 @@
 
 package controllers.agent
 
+import common.SessionKeys.viewedDDInterrupt
 import config.{AppConfig, ErrorHandler}
+import connectors.httpParsers.ResponseHttpParser.HttpResult
 import controllers.BaseController
 import controllers.predicates.AuthoriseAsAgentWithClient
 import javax.inject.{Inject, Singleton}
+import models.DirectDebit
+import models.errors.DirectDebitError
 import play.api.Logger
 import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
-import services.{CustomerDetailsService, DateService}
-import views.html.agent.AgentHubView
+import services.{CustomerDetailsService, DateService, DirectDebitService}
+import views.html.agent.{AgentHubView, DirectDebitInterruptView}
 
-import scala.concurrent.{ExecutionContext}
+import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
 class AgentHubController @Inject()(val authenticate: AuthoriseAsAgentWithClient,
                                    val serviceErrorHandler: ErrorHandler,
                                    val customerDetailsService: CustomerDetailsService,
                                    val dateService: DateService,
+                                   val directDebitService: DirectDebitService,
                                    mcc: MessagesControllerComponents,
                                    agentHubView: AgentHubView,
+                                   ddInterruptView: DirectDebitInterruptView,
                                    implicit val appConfig: AppConfig,
-                                   implicit val executionContext: ExecutionContext) extends BaseController(mcc) {
+                                   implicit val executionContext: ExecutionContext
+                                  ) extends BaseController(mcc) {
 
   def show: Action[AnyContent] = authenticate.async { implicit user =>
 
-      customerDetailsService.getCustomerDetails(user.vrn).map {
+    for {
+      customerInfo <- customerDetailsService.getCustomerDetails(user.vrn)
+      ddResult <-
+        if(appConfig.features.directDebitInterruptFeature()) {
```

Comment:
Can you extend this `if` condition so that it will only call the API when both the feature switch is on and there is no DD value in session.